### ISSUE TITLE
Mapping issue: Cannot migrate Related links

### ISSUE THREAD

**Issue by new2migration** [solidify/jira-azuredevops-migrator#905]
I am new to using the solidify migration tool and am having difficulty getting the related link types to export from jira and import to ADO for all issue types.  The parent and child work correctly but not the others.  We are still using the jira server NOT cloud version.  here is the code I have that does not seem to work -- what am I missing in my config file? NOTE: I did look up each one in jira under the advanced JCL to see if the source was correct and they appear to match the names i have listed below.  Could it be on the ADO side?

"link-map": {
    "link": [
      {
        "source": "Epic",
        "target": "System.LinkTypes.Hierarchy-Reverse"
      },
      {
        "source": "Parent",
        "target": "System.LinkTypes.Hierarchy-Reverse"
      },
      {
        "source": "Child",
        "target": "System.LinkTypes.Hierarchy-Reverse"
      },
      {
        "source": "Relates",
        "target": "System.LinkTypes.Related"
      },
      {
        "source": "Relates to",
        "target": "System.LinkTypes.Related"
      },
  
      { "source": "clones",
        "target": "System.LinkTypes.Related"
      },
      {
        "source": "is cloned by",
        "target": "System.LinkTypes.Related"
      },
      {
        "source": "blocks",
        "target": "System.LinkTypes.Related"
      },
      {
        "source": "is blocked by",
        "target": "System.LinkTypes.Related"
      },
      {
        "source": "is duplicated by",
        "target": "System.LinkTypes.Duplicate-Forward"
      },
      {
        "source": "duplicates",
        "target": "System.LinkTypes.Duplicate-Forward"
      }
    ]

**Comment by Alexander-Hjelm** [comment]
Here is the link map I usually use for Relates links:

```
      {
        "source": "Relates",
        "target": "System.LinkTypes.Related"
      },
```

Other than that, it could be that you either haven't mapped a particular link type, or maybe there are other errors in the migration log that need to be addressed first? This could cause some revisions to fail to import.

**Comment by new2migration** [comment]
@Alexander-Hjelm -- have you successfully been able to migrate the relationship types of clones or is cloned by?  If so would you be willing to share what code  you used for those relationship types as an example for me to see how those worked for you?

**Comment by Alexander-Hjelm** [comment]
Haven't tried myself, but I reckon it would be something like this:

```md
    {
    "source": "is cloned by",
    "target": "System.LinkTypes.Related"
  },   
  
  {
    "source": "clones",
    "target": "System.LinkTypes.Related"
  },
```

**Comment by psaunders2** [comment]
I am having the same issue, the links for the Epic migrate over fine but the liks for other ticket types do not. No errors occur but the links do not appear in the JIRA Export file.

"link-map": {
		"link": [{
				"source": "Epic",
				"target": "System.LinkTypes.Hierarchy-Reverse"
			},
			{
				"source": "Parent",
				"target": "System.LinkTypes.Hierarchy-Reverse"
			},
			{
				"source": "Child",
				"target": "System.LinkTypes.Hierarchy-Forward"
			},
			{
				"source": "Relates to",
				"target": "System.LinkTypes.Related"
			},
			{
				"source": "Duplicates",
				"target": "System.LinkTypes.Duplicate-Forward"
			},
                        {
				"source": "Clones",
				"target": "System.LinkTypes.Related"
			},
			{
				"source": "Relates to",
				"target": "System.LinkTypes.Related"
			},
.......

**Comment by psaunders2** [comment]
Thanks, I'll remove the duplicate relates to. I'll add Cloners too, thanks.

**Comment by psaunders2** [comment]
Thanks [new2migration]!!!  adding 'Cloners' worked to get them over, odd because we don't even use it in our JIRA, but whatever works.

It does appear I still have one issue, they come over BUT they do not come over as the correct 'link type', they are all 'Related'. SO I need to troubleshoot that now.

**Comment by new2migration** [comment]
@psaunders2 - There is no "cloned" or "blocked" type relationship in ADO so in the above we mapped ours to "related".  I would suggest reviewing the available link types in ADO and if related if not what is desired select one that exists that may work better.

**Comment by anurao98** [comment]
@Alexander-Hjelm, how should we map Causes and is caused by link types in tool v3.0.328 ?

**Comment by Alexander-Hjelm** [comment]
@anurao98 Please submit your support ticket as a new Issue.
